packages/smart-contracts/src/contracts/SingleRequestProxyFactory.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+import './ERC20SingleRequestProxy.sol';
+import './EthereumSingleRequestProxy.sol';
+
+/**
+ * @title SingleRequestProxyFactory
+ * @notice This contract is used to create SingleRequestProxy instances
+ */
+contract SingleRequestProxyFactory is Ownable {
+  address public ethereumFeeProxy;
+  address public erc20FeeProxy;
+
+  event EthereumSingleRequestProxyCreated(
+    address indexed proxyAddress,
+    address indexed payee,
+    bytes indexed paymentReference
+  );
```

Comment:
This may be true, but we should stick to the convention used by the existing payment proxies - all of which index the `paymentReference` in emitted events. Resolving comment as won't fix. :no_good: